### PR TITLE
[v5.x] Refactor tests

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR:
- [x] test harness updated to reflect removal of v4.x interpreter
- [x] more test cases added to cover possible MethodCalls in ErgoTree for some operations
- [x] more unused code removed

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/eval/IRContext.scala [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 import sigmastate.lang.TransformingSigmaBuilder
-import sigmastate.lang.exceptions.CostLimitException
-import sigmastate.utils.Helpers
-import sigmastate.utxo.CostTable
 
 import java.util.concurrent.locks.ReentrantLock
 import scala.util.Try
 
-trait IRContext extends Evaluation with TreeBuilding with GraphBuilding {
+trait IRContext extends TreeBuilding with GraphBuilding {
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/eval/IRContext.scala:8]: ScalaDoc missed

**Review by kushti — CHANGES_REQUESTED** [review]
Add ScalaDoc for `TreeBuilding` also.

`Evaluation` is not descriptive enough name I guess

Please specify in ScalaDoc comments whether classes belong to frontend compiler or tree interpreter.

File: sigmastate/src/main/scala/sigmastate/eval/IRContext.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/eval/IRContext.scala:8]: fixed

File: sigmastate/src/main/scala/sigmastate/eval/Evaluation.scala [hunk 2]
```diff
@@ -1,859 +1,24 @@
...
-      (y, costAccumulator.totalCost)
-    }
-    res
-  }
-}
+import java.math.BigInteger
+import scala.reflect.ClassTag
+import scala.util.Try
 
 object Evaluation {
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/eval/Evaluation.scala:19]: fixed. Also added TODO to either rename it or find a better place for the methods after code reorganization in next PRs.

**Comment by aslesarenko** [comment]
> Please specify in ScalaDoc comments whether classes belong to frontend compiler or tree interpreter.
@kushti in #827 and #833 the consensus code is moved into separate module which doesn't depend on  both compiler and graph-IR code. See `interpreter` and `sc` (aka sigma compiler) modules.

**Review by kushti — APPROVED** [review]

**Pull request merged**